src/js/static-net-filtering.js

```diff
@@ -3269,12 +3269,10 @@ const FilterParser = class {
 
     isJustOrigin() {
         return this.optionUnitBits === this.DOMAIN_BIT &&
-            this.isRegex === false && (
-                this.pattern === '*' || (
-                    this.anchor === 0b010 &&
-                    /^(?:http[s*]?:(?:\/\/)?)$/.test(this.pattern)
-                )
-            ) &&
+            this.isRegex === false &&
```

Comment:
ESLint can be difficult to please sometimes. Unfortunately the style used here is not supported, and I can't find a way to make it work with any configuration settings. If this doesn't work for you, I can try to figure it out a way.

Reply:
I have undone this part and also at https://github.com/gorhill/uBlock/pull/3800#discussion_r681720226

Reply:
This could be retwritten as:

```js
    isJustOrigin() {
        if ( this.optionUnitBits !== this.DOMAIN_BIT ) { return false; }
        if ( this.isRegex ) { return false; }
        if ( this.domainOpt.includes('~') ) { return false; }
        if ( this.pattern === '*' ) { return true; }
        if ( this.anchor !== 0b010 ) { return false; }
        if ( /^(?:http[s*]?:(?:\/\/)?)$/.test(this.pattern) ) { return true; }
        return false;
    }
```

Reply:
>This could be retwritten as …

If we rewrite any code then let's make it a separate patch because then we should also test the performance.